components/feature/sync/src/main/java/mozilla/components/feature/sync/FirefoxSyncFeature.kt

```diff
@@ -24,51 +24,94 @@ import kotlin.coroutines.CoroutineContext
  * (e.g. places and logins depend on native libraries), and we do not want to force a consumer which
  * only cares about syncing logins to have to import a places native library.
  *
- * @param coroutineContext A [CoroutineContext] that will be used to perform synchronization work.
  * @param reifyAuth A conversion method which reifies a generic [FxaAuthInfo] into an object of
  * type [AuthType].
  */
 class FirefoxSyncFeature<AuthType>(
```

Comment:
For the listeners here: Consider using `ObserverRegistry` (if applicable) which is used by some of our other classes. It takes care of dealign with listeners for you. More importantly it has hooks for binding subscriptions to a lifecycle/view or pausing them when the app is in the background. This avoids you having to deal with all those weak references and stuff in your app code: It will automatically pause or unsubscribe for you.